.github/workflows/build_test.yml

```diff
@@ -0,0 +1,238 @@
+        uses: docker/build-push-action@v6
+        with:
+          build-args: |
+            LD_FLAGS=-s -w -X github.com/Mirantis/hmc/internal/build.Version=${{ steps.vars.outputs.version }}
+          context: .
+          platforms: linux/amd64
+          tags: |
+            ghcr.io/mirantis/hmc/controller-ci:${{ steps.vars.outputs.version }}
+          push: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+      - name: Prepare and push HMC template charts
+        run: |
+          make hmc-chart-release
+          make helm-push
+
+  controller-e2etest:
+    name: E2E Controller
+    if: ${{ contains( github.event.pull_request.labels.*.name, 'skip for now') }}
+    runs-on: ubuntu-latest
+    needs: build
+    concurrency:
+      group: controller-${{ github.head_ref || github.run_id }}
+      cancel-in-progress: true
+    outputs:
+      clustername: ${{ needs.build.outputs.clustername }}
+      version: ${{ needs.build.outputs.version }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Setup kubectl
+        uses: azure/setup-kubectl@v4
+      - name: Run E2E tests
+        env:
+          GINKGO_LABEL_FILTER: 'controller'
+          MANAGED_CLUSTER_NAME: ${{ needs.build.outputs.clustername }}
+          IMG: 'ghcr.io/mirantis/hmc/controller-ci:${{ needs.build.outputs.version }}'
+          VERSION: ${{ needs.build.outputs.version }}
+        run: |
+          make test-e2e
+      - name: Archive test results
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: cloud-e2etest-logs
+          path: |
+            test/e2e/*.log
+
+  provider-cloud-e2etest:
+    name: E2E Cloud Providers
+    runs-on: ubuntu-latest
+    if: ${{ contains( github.event.pull_request.labels.*.name, 'test e2e') }}
+    needs: build
+    concurrency:
+      group: cloud-${{ github.head_ref || github.run_id }}
+      cancel-in-progress: true
+    outputs:
+      clustername: ${{ needs.build.outputs.clustername }}
+      version: ${{ needs.build.outputs.version }}
+    env:
+      AWS_REGION: us-west-2
+      AWS_ACCESS_KEY_ID: ${{ secrets.CI_AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.CI_AWS_SECRET_ACCESS_KEY }}
+      AZURE_SUBSCRIPTION_ID: ${{ secrets.CI_AZURE_SUBSCRIPTION_ID }}
+      AZURE_TENANT_ID: ${{ secrets.CI_AZURE_TENANT_ID }}
+      AZURE_CLIENT_ID: ${{ secrets.CI_AZURE_CLIENT_ID }}
+      AZURE_CLIENT_SECRET: ${{ secrets.CI_AZURE_CLIENT_SECRET }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Setup kubectl
+        uses: azure/setup-kubectl@v4
+      - uses: actions/checkout@v4
+      - name: Setup tmate session
```

Comment:
Should this be in here still?

Reply:
I'm still using it, I'm going to remove it once I get Azure green